src/mysql/language/rules/datatypes.ne

```diff
@@ -31,14 +31,18 @@ O_DATATYPE -> (
 #
 # https://mariadb.com/kb/en/data-types-numeric-data-types/
 
+# =============================================================
```

Comment:
Now I understand the misconception about the SQL concept of width you mentioned in the PR. I see the changes you proposed here.

I'm not worried about backwards compatibility in this case. I think it makes sense to place a breaking change (major version bump) to perform these fixes, as it changes a fix to the broken concept of "width" currently applied in the Compact JSON format. I suggest we remove `width` entirely and stick with `displayWidth` only.

Here's what I suggest:

```
O_INTEGER_DATATYPE ->
  (
      %K_INT        {% d => { return { datatype: d[0].value }} %}
    | %K_INTEGER    {% d => { return { datatype: d[0].value }} %}
    | %K_TINYINT    {% d => { return { datatype: d[0].value }} %}
    | %K_SMALLINT   {% d => { return { datatype: d[0].value }} %}
    | %K_MEDIUMINT  {% d => { return { datatype: d[0].value }} %}
    | %K_BIGINT     {% d => { return { datatype: d[0].value }} %}
  )
  ( _ %S_LPARENS _ %S_NUMBER _ %S_RPARENS {% d => d[3].value %} ):?

    {% d => {
      return {
        id: 'O_INTEGER_DATATYPE',
        def: {
          datatype: d[0].datatype,
          displayWidth: d[1]
        }
      }
    }%}
```

Reply:
Then in `src/mysql/formatter/compact/models/datatype.ts` file:

- replace `width` with `displayWidth` (also update related interfaces)
- add minimum and maximum properties so that in `fromDef` method, assign minimum and maximum, considering the calculation for that datatype.
- in `toJSON` and `clone` methods, assign minimum and maximum properties to the JSON.

Reply:
I prefer not using the new `x-` properties in the JSON schema. Minimum and maximum should be able to handle large numbers and they should be both parseable and stringifiable. What motivated you to use BigInt?

Reply:
<img width="540" alt="image" src="https://github.com/user-attachments/assets/662afc54-679e-444a-aa78-fbb8cd35b833" />

---------------------------------

the mysql int type ranges

https://dev.mysql.com/doc/refman/8.4/en/integer-types.html

<img width="1180" alt="image" src="https://github.com/user-attachments/assets/40ddaf21-37d0-4dce-a84d-0755822cfdc3" />

---------------------------------

1. BigInt - 2 ** 64: as you can see for the 1st picture, it is not precise for javascript to hold a huge number like 2 ^ 64, the bigger for the number, the accuracy would be losing using js, so for the more fine control for number, i would prefer BigInt to handle large numbers

2. I agree with your idea of replacing width with displayWidth, adding minimum and maximum properties, and in toJSON and clone methods, assign minimum and maximum properties to the JSON.

3. I am ok with not using the new x- properties in the JSON schema, but the issue for a big int in javascript would make the sql-ddl-to-json-schema not perfect with the big number in jsonschema, so it comes with the BigInt plan.

so that's why for the x- proposal

i'm just wanna give it an extra plan for who are using sql-ddl-to-json-schema to handle some big number

---------------------------------

so, what would you like to handling the big number

Reply:
ok I understand the motivation for using BigInt. Please make a proposal to add it to the compact JSON format under the x- properties but don't add it to the JSON schema yet. Let's first see how it looks. Thank you and happy new year 🎉

Reply:
Happy New Year for 2025 !

Reply:
## feat: update int datatypes with compact json and jsonschema

------------------------------

### Fix BigInt issue:

proposal for bigint in jsonschema instead of using BigInt in case jsonschema not be able to handle BigInt

1. max unsigned and max signed: Number.MAX_SAFE_INTEGE -> 9007199254740991

2. min signed: Number.MIN_SAFE_INTEGER -> -9007199254740991

3. min unsigned: 0

------------------------------

### features:

1. replace width with displayWidth (also update related interfaces)
2. add minimum and maximum properties so that in fromDef method, assign minimum and maximum, considering the calculation for that datatype.
3. in toJSON and clone methods, assign minimum and maximum properties to the JSON.

------------------------------

### fixes:

remove x- properties for bigint in the JSON schema